src/syscheckd/create_db.c

```diff
@@ -815,6 +741,32 @@ cJSON * fim_json_event(char * file_name, fim_entry_data * old_data, fim_entry_da
     cJSON_AddStringToObject(data, "type", FIM_EVENT_TYPE[type]);
     cJSON_AddNumberToObject(data, "timestamp", new_data->last_event);
 
+#ifndef WIN32
+    if (old_data != NULL) {
+        char** paths = NULL;
+
+        paths = fim_db_get_paths_from_inode(syscheck.database, old_data->inode, old_data->dev);
+
+        if(paths[1]){
```

Comment:
Please check both `paths[0]` and `paths[1]`.

We expect `paths` to contain one item at least. However, in a rare case where the file is removed by another thread —or a programming error—, `path` may contain 0 elements, so `paths[1]` remains undefined.